vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridDetailsRowIT.java

```diff
@@ -119,14 +118,38 @@ public void gridUpdateItemUpdateDetails() {
         assertElementHasNoButton(detailsElement.get(1));
         // detail on row 3 contains a button
         assertElementHasButton(detailsElement.get(2),"Person 3 - updates 1");
+    }
+
+    @Test
+    public void gridRemoveDetailsOpenedAttributeWhenItemIsDeleted() {
+        open();
+        GridElement grid = $(GridElement.class).first();
+
+        // select row 3 only
+        clickElementWithJs(getRow(grid, 2).findElement(By.tagName("td")));
```

Comment:
Initially I wanted to setup the test scenario using the two initially selected items from the page. However I ran into another bug, where selecting items on the server-side does not set the `details-opened` attribute at all.
So I had to override the initial selection on the client side for now, to get a row with a `details-opened` attribute.